### PR TITLE
FE-1207 - Remove the feature flag and remove the sending domain and tracking domain pages

### PR DESCRIPTION
**Pull request by shrirupa**
FE-1207 - Remove the feature flag and remove the sending domain and tracking domain pages from nav in hibana

### What Changed
 -  Removed the feature flag - allow_domains_v2
 - Removed Sending and Tracking Domain from navigation in Hibana
 - Removed the routes to sending domain and tracking domain in hibana
 - OG theme still has the old sending and tracking domain pages intact

### How To Test
 - Check on a account with allow_domain_v2 not enabled, domains page is visible in the nav
 - Check in OG version the sending and tracking domain pages are visible

### To Do
- [ ] Address feedback

### CODE REVIEW

File: cypress/fixtures/account/200.get.has-domains-v2.json [hunk 1]
```diff
@@ -5,7 +5,6 @@
     },
     "options": {
       "ui": {
-        "allow_domains_v2": true
       }
     }
   }
```
Comment (nicklemmon) [inline cypress/fixtures/account/200.get.has-domains-v2.json:10]: I think this whole fixture can be removed? (and any references to it)

**Review by nicklemmon — CHANGES_REQUESTED** [review]
@shrirupa Just had one tiny thing - we can remove that fixture from the repo entirely (and any references to it). Everything else looks good.

**Comment by shrirupa** [comment]
@nicklemmon Thanks for the feedback! This is ready for another pass.

I have added a few more things to the PR - 
* Replaced the Select in Filters with ListBox (UX wanted it to be a ListBox instead of Select)
* Added a component wrapper for ListBox
* Fixed the spacing bug that you found.

File: src/pages/domains/components/TableFilters.js [hunk 2]
```diff
@@ -95,13 +95,17 @@ function SortSelect({ options, onChange, disabled }) {
 
   return (
     <StyledGridCell>
-      <Select
+      <ListBox
```
Comment (nicklemmon) [inline src/pages/domains/components/TableFilters.js:98]: The keyboard controls for this aren't working? They work here (focus with your keyboard, hit the "down" arrow and you can select an option)
https://design.sparkpost.com/components/listbox

Not sure if their example is set up with Downshift.

File: cypress/tests/integration/domains/listPage.spec.js [hunk 3]
```diff
@@ -288,7 +288,8 @@ describe('The domains list page', () => {
         cy.visit(PAGE_URL);
         cy.wait(['@sendingDomainsReq', '@subaccountsReq']);
 
-        cy.findByLabelText('Sort By').select('Domain Name (A - Z)');
+        cy.findByLabelText('Sort By').click();
+        cy.findAllByText('Domain Name (A - Z)').click();
```
Comment (nicklemmon) [inline cypress/tests/integration/domains/listPage.spec.js]: You should be able to use `cy.findByRole('option',  { name: 'Domain Name (A - Z)' }).click()`

File: src/components/matchbox/ListBox.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  if (!isHibanaEnabled) {
+    throw new Error(
+      'ListBox component not available in original matchbox. Please remove or restrict to Hibana',
+    );
+  }
+
+  return <HibanaListBox {...props} />;
+}
+
+function Option(props) {
```
Comment (nicklemmon) [inline src/components/matchbox/ListBox.js:18]: I think we need to forward refs for this wrapper:
Screen Shot 2020-11-10 at 2 15 22 PM

**Review by nicklemmon — CHANGES_REQUESTED** [review]
@shrirupa A couple of things - mainly related to the `ListBox`. I think the keyboard controls might not be working because of the lack of ref forwarding?

There are also a few errors in the console I'm seeing you should take a look at:
Screen Shot 2020-11-10 at 2 18 07 PM
Screen Shot 2020-11-10 at 2 18 07 PM
Screen Shot 2020-11-10 at 2 18 45 PM
Screen Shot 2020-11-10 at 2 19 06 PM

File: src/components/matchbox/ListBox.js (hunk 4 above)
Comment (shrirupa) [reply src/components/matchbox/ListBox.js:18]: Thanks for catching this!

**Comment by shrirupa** [comment]
@nicklemmon Thanks again for catching that forwardRef issue. This is ready for another pass.

**Review by nicklemmon — APPROVED** [review]
@shrirupa Looks good! The keyboard controls are working now for the `ListBox`. Nice job.

Let me know when you want to deploy.

**Pull request merged**